src/v2/Apps/Show/ShowApp.tsx

```diff
@@ -38,7 +38,7 @@ const FullScreenSeparator = styled(Separator)`
 export const ShowApp: React.FC<ShowAppProps> = ({ show }) => {
   const { contextPageOwnerSlug, contextPageOwnerType } = useAnalyticsContext()
 
-  const hasViewingRoom = show.viewingRoomIDs.length > 0
+  const hasViewingRoom = show.viewingRoomsConnection?.edges.length > 0
```

Comment:
It's a bit surprising to me that `show.viewingRoomsConnection` can be null but I got the following error message while rendering a show:

```
Error Message: Cannot read property 'edges' of null
```

http://localhost:5000/show/gallery-fifty-one-malkovich-malkovich-malkovich-homage-to-photographic-masters

Reply:
Yea, I guess this field really is nullable (based on the GraphQL types), and it would be `null` when there are no viewing rooms linked. The type is https://github.com/artsy/metaphysics/blob/890f2784de11b8225e2b4392119da0b059c544b6/src/lib/stitching/gravity/stitching.ts#L130 (lacking the `!`), and https://github.com/artsy/metaphysics/blob/890f2784de11b8225e2b4392119da0b059c544b6/src/lib/stitching/gravity/stitching.ts#L540 (resolver).

We could update those to be non nullable w/ an empty connection when there is none set, but technically for now Force needs the safe `?` operator.

Personally it doesn't bother me to have fields be nullable and need to use `?` more, or avoid deeply destructuring incoming props, etc. But, with turning the strict null checks on, we see that it can actually become really unwieldy in practice (i think?), so I'd be in favor of refactoring this type to truly be non-nullable (every little bit helps 😅 )

Reply:
Malkovich!